executor.go

```diff
@@ -50,14 +51,45 @@ func (x *executor) Execute(req *Request, re ResponseEmitter, env Environment) er
 			return err
 		}
 	}
+
+	return EmitResponse(cmd.Run, req, re, env)
```

Comment:
Sorry, I just realized we're doing this in the wrong place. `Execute` should just execute the command (pre, run, post) and feed the results to the response emitter.

I believe the best place to put it is in `cli.NewResponseEmitter`. That is, if the command defines a `DisplayCLI` function, `cli.NewResponseEmitter` would:

1. Call `NewChanResponsePair` to get an emitter/response.
2. Pass the response to `DisplayCLI`.
3. Return the emitter (you may need to wrap it in something? not sure).

That way, we only attempt to write to the terminal if we're actually running in a CLI.

Reply:
Followed this approach.